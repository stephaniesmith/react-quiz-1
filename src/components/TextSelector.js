import React from 'react';
import PropTypes from 'prop-types';

const TextSelector = ({ text, onChange }) => {
  return (
    <div>
      <label htmlFor="text">Text</label>
      <input type="text" name="text" value={text} onChange={onChange}/>
    </div>
  );
};

TextSelector.propTypes = {
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string
};

export default TextSelector;
