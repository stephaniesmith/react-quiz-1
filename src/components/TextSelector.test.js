import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector', () => {
  it('renders TextSelector', () => {
    const wrapper = shallow(<TextSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
