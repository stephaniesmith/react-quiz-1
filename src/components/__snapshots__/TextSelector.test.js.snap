// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
<div>
  <h1>
    App
  </h1>
  <TextSelector
    onChange={[Function]}
    text="Text Here"
  />
  <ColorSelector
    onChange={[Function]}
  />
  <Shape
    backgroundColor="black"
    color="white"
    text="Text Here"
  />
</div>
`;

exports[`TextSelector renders TextSelector 1`] = `
<div>
  <label
    htmlFor="text"
  >
    Text
  </label>
  <input
    name="text"
    type="text"
  />
</div>
`;
